### PR TITLE
server/scheduler: check whether is valid for config of hot scheduler …

### PR DESCRIPTION
**Pull request by njuwelkin**
…in http

After https://github.com/tikv/pd/pull/3920, we only check config for config of hot scheduler in pd-ctl, we also need check it in http.

Issue Number: Close tikv/pd#3952

### What is changed and how does it work?
Added a method 'validPriority' for structure hotRegionSchedulerConfig.
when receiving a http request to config hot region scheduler, we make below validations:
- whether scheduling dimension is valid.
- whether have at least two priorities
- whether the priorities are repeated
- whether the priority is set to QueryPriority for write-peer-priorities

```commit-message
```

### Check List
Tests
- Unit test

### Release note
None

### CODE REVIEW

File: server/schedulers/hot_region_config.go [hunk 1]
```diff
@@ -299,6 +301,38 @@ func (conf *hotRegionSchedulerConfig) handleGetConfig(w http.ResponseWriter, r *
 	rd.JSON(w, http.StatusOK, conf.getValidConf())
 }
 
+func (conf *hotRegionSchedulerConfig) validPriority() error {
+	isValid := func(priorities []string) (map[string]bool, error) {
+		priorityMap := map[string]bool{}
+		for _, p := range priorities {
+			if p != BytePriority && p != KeyPriority && p != QueryPriority {
+				return nil, errs.ErrSchedulerConfig.FastGenByArgs("Invalid scheduling dimensions.")
```
Comment (lhy1024) [inline server/schedulers/hot_region_config.go:309]:
```suggestion
				return nil, errs.ErrSchedulerConfig.FastGenByArgs("invalid scheduling dimensions.")
```

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

**Comment by lhy1024** [comment]
plz sign commit, and let me know if you encounter any problems

**Comment by njuwelkin** [comment]
there were two commits, and the 1st was not signed. so I merged two commits to one, in order to sign the 1st one.

**Review by lhy1024 — APPROVED** [review]

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**